src/mrpro/utils/_databuffermixin.py

```diff
@@ -0,0 +1,104 @@
+from collections import OrderedDict
+
+import torch
+
+from mrpro.data._Data import Data
+
+
+class DataBufferMixin(torch.nn.Module):
+    """A Mixin that allow to set a mrpro.data.Data as a Buffer.
+
+    This implements the same logic for mrpro.Data as torch.nn.Module
+    uses for Buffers, but stores them in a separate dictionary called
+    _data
+
+    The main use is to allow Data objects to be automatically moved to a
+    devices etc if one calles, for example DataBufferMixin.cuda()
+
+    Used in Operators, for example.
+    """
+
+    _data: dict[str, Data]
+    # could maybe be requiered at some point,
+    # see torch.nn.Module
+    # call_super_init = True
+
+    def register_buffer(self, name: str, data: torch.Tensor | None | Data, persistent: bool = True) -> None:
+        r"""Add a buffer to the module.
+
+        This is typically used to register a buffer that should not to be
+        considered a model parameter.
+        Buffers, by default, are persistent and will be saved alongside parameters. This
+        behavior can be changed by setting :attr:`persistent` to ``False``. The
+        only difference between a persistent buffer and a non-persistent buffer
+        is that the latter will not be a part of this module's
+        :attr:`state_dict`.
+
+        Buffers can be accessed as attributes using given names.
+
+        See also torch.nn.Module.register_buffer.
+
+        Parameters
+        ----------
+            name
+                name of the buffer. The buffer can be accessed from this module using the given name
+            data
+                buffer to be registered. If ``None``, then operations that run on buffers, such as `cuda`, are ignored.
+                If ``None``, the buffer is **not** included in the module's `state_dict`.
+            persistent (bool)
+                whether the buffer is part of this module's `state_dict`.
+        """
+        if not isinstance(data, Data):
+            return super().register_buffer(name, data, persistent)
+
+        if '_data' not in self.__dict__:
+            raise AttributeError('cannot assign buffer before super().__init__() call')
+        elif not isinstance(name, str):
+            raise TypeError(f'buffer name should be a string. Got {torch.typename(name)}')
+        elif '.' in name:
+            raise KeyError("buffer name can't contain \".\"")
```

Comment:
can't -> cannot otherwise formatting of error message is not nice

Reply:
This is the same message pytorch prints. I'd prefer to keep it as is

Reply:
Just because pytorch does it does not mean it is correct ;). Keep it if you want, I don't feel strongly about it, just noticed it when I tested the errors.